src/encord_active/app/project_comparison/project_similarity.py

```diff
@@ -0,0 +1,165 @@
+from typing import cast
+
+import numpy as np
+import pandas as pd
+import streamlit as st
+from scipy.stats import ks_2samp
+
+from encord_active.app.common.state import get_state
+from encord_active.app.common.utils import setup_page
+from encord_active.app.projects_page import project_list
+from encord_active.lib.charts.project_similarity import (
+    ProjectSimilaritySchema,
+    plot_project_similarity_metric_wise,
+    render_2d_metric_similarity_plot,
+)
+from encord_active.lib.db.connection import DBConnection
+from encord_active.lib.db.merged_metrics import MergedMetrics
+from encord_active.lib.metrics.metadata import fetch_metrics_meta
+from encord_active.lib.project import ProjectFileStructure
+from encord_active.lib.project.metadata import fetch_project_meta
+
+# The following metrics does not give any information comparable to another project
+METRICS_TO_EXCLUDE = [
+    "Random Values on Objects",
+    "Random Values on Images",
+    "Image Difficulty",
+]
+
+
+def calculate_metric_similarity(array_1: np.ndarray, array_2: np.ndarray) -> float:
+    ks_score, _ = ks_2samp(array_1, array_2)
+    return 1 - ks_score
+
+
+def render_2d_metric_similarity_container(
+    all_metrics: list, merged_metrics_1: pd.DataFrame, merged_metrics_2: pd.DataFrame, project_name_2: str
+):
+    metrics_filtered = []
+    for metric_name in all_metrics:
+        if metric_name in merged_metrics_1 and merged_metrics_1[metric_name].dropna().shape[0] > 0:
+            metrics_filtered.append(metric_name)
+
+    if len(metrics_filtered) <= 1:
+        st.write("At least two metrics should be generated for 2D metric similarity plot")
+        return
+
+    metric_selection_col_1, metric_selection_col_2 = st.columns(2)
+    metric_name_1 = metric_selection_col_1.selectbox(
+        "Select the first metric",
+        options=metrics_filtered,
+        index=0,
```

Comment:
Would be nice to default to the two least similar metrics such that the plot looks most interesting from the beginning.

Reply:
👍 Put Brightness vs. contrast for now. They both seem good. Unbounded metrics like Area etc. seem not attractive I think.